game/theater/iadsnetwork.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import List, Dict, Any, Optional, Union
+
+from game.theater.theatergroundobject import (
+    IadsGroundObject,
+    IadsBuildingGroundObject,
+    SamGroundObject,
+    EwrGroundObject,
+    TheaterGroundObject,
+    IADSRole,
+)
+
+
+def find_tgo_by_original_name(
+    name: str, ground_objects: List[TheaterGroundObject[Any]]
+) -> List[TheaterGroundObject[Any]]:
+    return [
+        go
+        for go in ground_objects
+        if isinstance(go, IadsGroundObject)
+        or isinstance(go, IadsBuildingGroundObject)
+        and go.original_name == name
+    ]
```

Comment:
Should be placed in ()

`if (isinstance(go, IadsGroundObject)
         or isinstance(go, IadsBuildingGroundObject))
         and go.original_name == name `